pipeline/DESIGN.md

```diff
@@ -0,0 +1,190 @@
+| stream_type | u8                | Cid of the stream        |
+| controller  | string            | Controller of the stream |
+| dimensions  | map(string,bytes) | Dimensions of the stream |
+
+## chain_proofs
+
+The chain_proofs table contains a row for each on chain proof.
+
+TBD how this table is populated and its schema.
+
+## conclusion_events
+
+The conclusion_events table contains a row for each event in a stream and represents a raw input event after various conclusions have been made.
+
+### Features
+
+* Access to events allowing in order access to event within stream
+* Access to pre-aggregated data for users building their own aggregation system
+* Validation of the event signatures
+
+### Schema
+
+| Column      | Type              | Description                                                                                     |
+| ------      | ----              | -----------                                                                                     |
+| index       | u64               | Order of this event. Index is always greater than the index of any previous event in the stream |
+| stream_cid  | bytes             | Cid of the stream                                                                               |
+| stream_type | u8                | Stream type                                                                                     |
+| controller  | string            | Controller of the stream                                                                        |
+| dimensions  | map(string,bytes) | Set of key values dimension pairs of the stream                                                 |
+| event_cid   | bytes             | Cid of the event                                                                                |
+| event_type  | u8                | One of Data (0) or Time (1)                                                                     |
+| data        | bytes             | DAG-JSON encoding of the event payload                                                          |
+| previous    | list(bytes)       | List of CID previous to this event                                                              |
+
+
+### Transformation
+
+Raw events are transformed into a flattened structure where stream dimensions and conclusions about data are added to the data.
+
+Conclusions include:
+
+* The event has a valid signature
+* The dimensions and controller of the event
+
+This table joins the raw_events, chain_proofs, and streams tables in order to make the conclusions about the raw events.
+
+## model_schemas
+
+The model_schemas table contains a row for each model and contains the complete resolved schema of the model.
+
+TBD how this table is populated and its schema.
+This table may be able to be combined with the streams table. Should we?
+
+## event_states
+
+The event_states table contains a row for each event in a stream and the state of the document at that point in the stream.
```

Comment:
I think adding more context to "state" here might be useful but I'm not totally sure what I want/expect. Maybe something about how the state determined by applying the  stream operation (currently JSON patch) to arrive at the materialized state.. I don't know, maybe not helpful. 

I'm trying to decide if this should be accessible to newer ceramic consumers and provide a lot of context or be more of a guide for us. Some of my other comments are also along that line and might not apply.

Reply:
The audience of this doc are developers of ceramic-one. For now I'll leave the details of these other bits light. However this does bring up some good content for a similar doc we will want on the docs sites.